docs/design/configs.md

```diff
@@ -0,0 +1,97 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
+
+## Problem
+
+Currently command-line flags and environment variables are used for component configurations in ``Kubernetes``.
+But, this approach is not suitable for a distributed computing environment. Specifically the 
+problems with the current approach are:
+
+* Hard to synchronize across replicated components.
+* Hard to change dynamically as needed by continuous services.
+* Hard to change in different versions that may support different configuration options.
+
+## Solution
+
+A new ``Config`` resource is proposed to solve the aforementioned problems 
+that is stored in ``etcd`` and distributed to interested/necessary components 
+through the ``APIServer``. ``Config`` resource will be mounted as a new volume 
+and will be managed by a volume plug-in that handles the ``APIServer`` communications.
+
+### Advantages
+
+* Reusable across different components.
+* Easy distribution thanks to ``etcd``, a highly-available KVL store.
+* By leveraging the power of ``/watch API`` along with the new resource, we can change component configurations dynamically.
+* Layer of abstraction that gets rid of the global state currently used for command-line flags and environment variables.
+
+### API Resource
+
+A new resource for ``Config`` will be added to the ``API``:
+
+```go
+    type Config struct {
+        TypeMeta   `json:",inline" yaml:",inline"`
+        ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+        // Flags contains command-line flags.
+        Flags     map[string]string `json:"flags,omitempty" yaml:"flags"`
```

Comment:
map[string]string is just a placeholder. I wasn't sure what would be needed and how to structure it(enum or struct). But, the end version will be very similar to a Container type.

Yes, Kubelets will be applying the Config object.